astropy/table/column.py

```diff
@@ -1142,7 +1142,7 @@ def _encode_str(value):
                 arr = np.char.encode(arr, encoding="utf-8")
                 if isinstance(value, np.ma.MaskedArray):
```

Comment:
This would fail for `Masked` arrays. How about just looking for a masked property, e.g., `if getattr(value, "mask", False) is not False`

Reply:
If you open an issue, please cross-link to this convo here. Thanks!